File: frontend/service/spareparts/catalog/CategoryService.ts
import {Category, ProcessCategory} from "types/spareparts/catalog/category";
import {Axios} from "core/axios";
import {IncomingHttpHeaders} from "http";

export class CategoryService {

  private static readonly URL = '/category';

  static async getAll(headers?: IncomingHttpHeaders): Promise<Category[]> {
    const {data: categories} = await Axios.get<Category[]>(this.URL, {headers});
    return categories;
  }

  static async add(processCategory: ProcessCategory): Promise<Category> {
    const {data: category} = await Axios.put<Category>(this.URL, processCategory);
    return category;
  }

  static async update(id: number, processCategory: ProcessCategory): Promise<Category> {
    const {data: category} = await Axios.patch<Category>(`${this.URL}/${id}`, processCategory);
    return category;
  }

  static async delete(id: number | string): Promise<Category> {
    const {data: category} = await Axios.delete<Category>(`${this.URL}/${id}`);
    return category;
  }

}

File: backend/src/main/java/by/minilooth/diploma/service/catalog/CategoryService.java
package by.minilooth.diploma.service.catalog;

import by.minilooth.diploma.exception.ActionIsImpossibleException;
import by.minilooth.diploma.exception.categories.CategoryNotFoundException;
import by.minilooth.diploma.models.bean.catalog.Category;
import by.minilooth.diploma.models.spareparts.catalog.ProcessCategory;

import java.util.List;
import java.util.Optional;

public interface CategoryService {

    void save(Category category);
    Category save(ProcessCategory processCategory);
    Category update(ProcessCategory processCategory, Long id) throws CategoryNotFoundException;
    void delete(Category category);
    Category delete(Long id) throws CategoryNotFoundException, ActionIsImpossibleException;
    Category getById(Long id) throws CategoryNotFoundException;
    List<Category> getAll();

}
